fix empty response from NLU

rameshpoomalai opened the pull request:
updated the code to display error message when there is no response from NLU/Error

rameshpoomalai committed:
fix empty response from NLU

stevemar commented on src/main/java/com/ibm/watson/nlu/DemoServlet.java:
```diff
@@ -89,65 +87,31 @@ protected void doPost(final HttpServletRequest req, final HttpServletResponse re
 			logger.log(Level.SEVERE, "Service error: " + e.getMessage(), e);
-			resp.getWriter().println("{ \"language\": \"en\", \"entities\": [ { \"type\": \"Error\", \"text\": \"" +
-					e.toString() + "\", \"count\": 1 } ] }");
+            resp.sendError(HttpStatus.SC_BAD_GATEWAY, e.getMessage() );
```
the indenting here seems off to me

rameshpoomalai replied:
Sorry,Let me fix the indent issue

rameshpoomalai committed:
fixed the indents

rhagarty reviewed (changes requested):
This change breaks running locally.
'getConfigParams()' is required to pull the user entered creds and modelID from config.properties.

rameshpoomalai committed:
Reverted the code which is lost during the previous checkin

Reverted the code which is lost during the checkin
Updated the code to handle empty/error response from NLU Service

rameshpoomalai committed:
Fixed the indents

rameshpoomalai commented:
Hi Rich, I have. merged the content which is lost during last checkin and fixed the error handling for short text. Can you please review it?

rhagarty commented:
Hello @rameshpoomalai - tested this locally and I don't like the error messages. First off, always displaying "Error 502" doesn't seem very helpful. Secondly, the error for bad model ID is no obvious what the problem is.

Here is a comparison of the errors shown for bad credentials and model ID not found:

Previous Version -
Error:com.ibm.watson.developer_cloud.service.exception.UnauthorizedException: Unauthorized: Access is denied due to invalid credentials
Error:com.ibm.watson.developer_cloud.service.exception.NotFoundException: model not found

Your proposed version - 
Error 502: Unauthorized: Access
Error 502: {extract.model_id.max_length}

stevemart commented:
@rhagarty can you propose messages that you would like to see?

rhagarty commented:
I would prefer keeping the existing messages. This PR, however, does handle an additional error case - server not responding. I think that part of the PR needs to be merged.

@markstur - thoughts?

rameshpoomalai committed:
Updated the error message format

rameshpoomalai commented:
@rhagarty ,I have updated the code to Match with the previous message format. Last time I was printing only the error message and removed the package details, because that expose the API details. Now I have matched the format with previous version and delivered.  

rhagarty commented:
@rameshpoomalai - the error for invalid credentials looks good, but the one for invalid model ID still needs some work. It shows as "Error:com.ibm.watson.developer_cloud.service.exception.InternalServerErrorException: : {extract.model_id.max_length}

Can you please change this to "Error:com.ibm.watson.developer_cloud.service.exception.NotFoundException: model not found"?

rameshpoomalai commented:
Hi Rich,
 It looks you are testing with the model with more than the accepted length. if i copy and past same id one more time i could reproduce this error, otherwise i am getting following error.
"Error:com.ibm.watson.developer_cloud.service.exception.NotFoundException: model not found ",

if required i can add validation for model length and handle it, let me know if required?

Thanks
Ramesh

rhagarty commented:
@rameshpoomalai - yes, you are correct - I had too many chars in my ID. A bad ID gives me the correct error. Thanks for your patience on this.
